### DISCUSSION TITLE
Problem with stream

### DISCUSSION THREAD

**Discussion by AkramZerarka** [livewire/livewire#8813] [7. Bugs (archived)]
### Livewire version

v3.5.8

### Laravel version

v10.48.20

### Which PHP version are you using?

PHP 8.3

### Description:
When using the `HasKheops` trait, there is an issue where a `SyntaxError: Unexpected token '<', "<!DOCTYPE "... is not valid JSON` is thrown during a streaming request.

This occurs when the following conditions are met:
- The `askKheops` method is executed to handle LLM (Large Language Model) responses.
- Upon completion of streaming data, instead of receiving valid JSON, the system tries to parse an HTML response (likely an error page), resulting in a syntax error.

### Error Stack:
VM6546:1 Uncaught (in promise) SyntaxError: Unexpected token '<', "<!DOCTYPE "... is not valid JSON at JSON.parse (<anonymous>) at sendRequest (livewire.js?id=cc800bf4:4321:52) at async RequestPool.send

### Steps to Reproduce:
1. Call the `askKheops` method.
2. Stream data from the LLM through `HandlesStreaming`.
3. Wait for the stream to complete.
4. Observe the console error stating a JSON parsing issue.

### Expected Behavior:
The stream should complete without attempting to parse non-JSON data, or the error should be handled gracefully if the response is not valid JSON.

### Code Sample:
```php
public function askKheops($store = false): void
{
    $llm = LlmManagerFacade::build('kheops');
    $message = $this->pull('generatedMessage');
    foreach ($llm->getIteration($message) as $item) {
        if (isset($item['error'])) {
            try {
                $this->closeModal();
            } catch (Exception $exception) {
                Log::error($exception->getMessage());
            }
            $this->notification()->error(
                $title = trans('Server Error'),
                $description = $item['error']
            );
        } else {
            $this->dispatch('scrollToBottom');
            $this->stream('content', nl2br($item));
        }
    }
    $this->content = $llm->getGeneratedText();

    if ($store) {
        $this->store(message: $this->content);
        $this->reset('isLoading');
    }
}


### Repository or Wirebox URL

https://wirebox.app/b/430z7

### Additional information

_No response_

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by boptom** [comment]
For anyone who comes across this, it may be related to: Maximum execution time of 30 seconds exceeded

Change or add this line in php.ini:
max_execution_time = 300

**Comment by david371571** [comment]
Anyone else run into this?

I have an old project (started on Laravel 8 + Livewire 2, now upgraded to Laravel 12 + Livewire 3) where wire:stream causes a Unexpected end of JSON input error at the end of the stream. In a fresh Laravel 12 + Livewire 3 project, the same code works fine.

In the working project, the response ends with a {"components":[...]}
block. In the broken one, it ends after the last {"stream":true,...} chunk and never sends the components block.

I removed all middlewareand dev tools, but nothing helps. Anyone know what might cause Livewire to skip the final response?

**Reply by boptom** [reply]
Did you try the above?

https://github.com/livewire/livewire/discussions/8813#discussioncomment-11653233

**Reply by david371571** [reply]
Found it... It's a symfony/http-foundation bug introduced in v7.3. See https://github.com/laravel/framework/issues/55894
Solved by adding v7.2 to composer.json:

```json
{
    "require": {
        ...
        "symfony/http-foundation": "~7.2.0"
        ...
    }
}
```
